### DISCUSSION TITLE
How to ensure SwiftBar uses right version of python

### DISCUSSION THREAD

**Discussion by arohl** [swiftbar/SwiftBar#305] [Q&A]
I want my python script to use the python installed through asdf. So I think I have two questions

1. What PATH does SwiftBar use? - I don't think it is my PATH as the logged in user
2. I use SwiftBar on my home and work machines and I have different user names on the two machines. I have everything working at the moment by specifying a path via env on mu shebang line but I have to specify two different paths so it will work on both machines. Is there an easier way to do this?

Many thanks

  Andrew

**Comment by melonamin** [comment]
> What PATH does SwiftBar use? - I don't think it is my PATH as the logged in user

The answer is - I'm not entirely sure...

By default, SwiftBar will launch your script like this:
`/bin/shell -c -l <script path>`

`shell` is either zsh or bash, depending on what you've selected in Preferences. Option '-l' should give us a login shell, meaning all relevant zsh configuration files are loaded before script execution. More [here](https://apple.stackexchange.com/questions/388622/zsh-zprofile-zshrc-zlogin-what-goes-where)

Meanwhile in my testing I see that `$PATH` in SwiftBar is different from `$PATH` in terminal, they are almost the same but asdf is defenitly missing.

**Comment by kdeldycke** [comment]
After 5 years maintaining a Python plugin for Xbar and SwiftBar, I ended up with this strategy: https://github.com/kdeldycke/meta-package-manager/blob/5e34d43932f562a65d68a1702c757deb6db2a43a/meta_package_manager/bar_plugin/meta_package_manager.7h.py#L111-L154
